packages/jsapi-utils/src/TableUtils.test.ts

```diff
@@ -523,6 +523,91 @@ describe('quick filter tests', () => {
       (Date.parse(dateString) as unknown) as DateWrapper;
   });
 
+  describe('escapeQuickTextFilter', () => {
+    // test case, expected result
+    test.each([
+      ['', ''],
+      [' ', ' '],
+      [null, null],
+      ['test', 'test'],
+      ['test\\test', 'test\\test'],
+      ['null\\null', 'null\\null'],
+      ['\\null\\null', '\\null\\null'],
+      ['=test', '\\=test'],
+      ['\\=test', '\\=test'],
+      ['\\test', '\\test'],
+      ['null', '\\null'],
+      ['NULL', '\\NULL'],
+      ['\\null', '\\\\null'],
+      ['\\testnull', '\\testnull'],
+      ['\\nulll', '\\nulll'],
+      ['\\\\null', '\\\\\\null'],
+      ['=null', '\\=null'],
+      ['!null', '\\!null'],
+      ['!=null', '\\!=null'],
+      ['*fish', '\\*fish'],
+      ['\\*fish', '\\*fish'],
+      ['=*fish', '\\=*fish'],
+      ['shooting*', 'shooting\\*'],
+      ['shooting\\*', 'shooting\\*'],
+      ['!=shooting*', '\\!=shooting*'],
+      ['==', '\\=='],
+      ['=', '\\='],
+      ['!=', '\\!='],
+      ['!~', '\\!~'],
+      ['~', '\\~'],
+      ['!', '\\!'],
+    ])(
+      'given %p, escape quick filter should return %p',
+      (testCase, expectedResult) => {
+        expect(TableUtils.escapeQuickTextFilter(testCase)).toBe(expectedResult);
+      }
+    );
+  });
+
+  describe('unescapeQuickTextFilter', () => {
```

Comment:
This is just the flip side, right? You could just add another expect in the previous test.each, might make it more clear that we expect this to flip and flip back.
```
      'given %p, escape quick filter should return %p',
      (unescaped, escaped) => {
        expect(TableUtils.escapeQuickTextFilter(unescaped)).toBe(escaped);
        expect(TableUtils.unescapeQuickTextFilter(escaped)).toBe(unescaped);
      }
```

Reply:
There's a couple of unique tests either side, for checking already escaped and already unescaped strings,  I could consolidate the one that are equal though.